src/DetailsView/components/left-nav/get-selected-assessment-summary-model.ts

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { DetailsViewPivotType } from 'common/types/store-data/details-view-pivot-type';
+import {
+    GetAssessmentSummaryModelFromProviderAndStatusData,
+    GetAssessmentSummaryModelFromProviderAndStoreData,
+} from 'reports/get-assessment-summary-model';
+import {
+    GetQuickAssessSummaryModelFromProviderAndStatusData,
+    GetQuickAssessSummaryModelFromProviderAndStoreData,
+} from 'reports/get-quick-assess-summary-model';
+
+export type GetSelectedAssessmentSummaryModelFromProviderAndStatusData =
+    | GetQuickAssessSummaryModelFromProviderAndStatusData
+    | GetAssessmentSummaryModelFromProviderAndStatusData;
+
+export type GetSelectedAssessmentSummaryModelFromProviderAndStoreData =
+    | GetAssessmentSummaryModelFromProviderAndStoreData
+    | GetQuickAssessSummaryModelFromProviderAndStoreData;
+
+export type SummaryModelGettersTypeMap = {
+    getAssessmentSummaryModelFromProviderAndStoreData: GetSelectedAssessmentSummaryModelFromProviderAndStoreData;
+    getAssessmentSummaryModelFromProviderAndStatusData: GetSelectedAssessmentSummaryModelFromProviderAndStatusData;
+};
+
+export type AssessmentSummaryModelGetters = {
+    [key in
+        | DetailsViewPivotType.assessment
+        | DetailsViewPivotType.mediumPass]: SummaryModelGettersTypeMap;
+};
+
+export type GetSelectedAssessmentSummaryModelGettersProps = {
+    getAssessmentSummaryModelFromProviderAndStoreData: GetSelectedAssessmentSummaryModelFromProviderAndStoreData;
+    getQuickAssessSummaryModelFromProviderAndStoreData: GetSelectedAssessmentSummaryModelFromProviderAndStoreData;
+    getAssessmentSummaryModelFromProviderAndStatusData: GetSelectedAssessmentSummaryModelFromProviderAndStatusData;
+    getQuickAssessSummaryModelFromProviderAndStatusData: GetSelectedAssessmentSummaryModelFromProviderAndStatusData;
+};
+
+export type SelectedAssessmentSummaryModelGetter =
+    | GetSelectedAssessmentSummaryModelFromProviderAndStoreData
+    | GetSelectedAssessmentSummaryModelFromProviderAndStatusData;
+
+export type GetSelectedAssessmentSummaryModelGetters = (
+    props: GetSelectedAssessmentSummaryModelGettersProps,
+) => SummaryModelGettersTypeMap;
+
+export const getSelectedSummaryModelGettersForAssessment: GetSelectedAssessmentSummaryModelGetters =
+    (props: GetSelectedAssessmentSummaryModelGettersProps) => {
+        return getSelectedSummaryModelGetters(props)[DetailsViewPivotType.assessment];
+    };
+
+export const getSelectedSummaryModelGettersForQuickAssess = (
+    props: GetSelectedAssessmentSummaryModelGettersProps,
+) => {
+    return getSelectedSummaryModelGetters(props)[DetailsViewPivotType.mediumPass];
+};
+
+const getSelectedSummaryModelGetters = (props: GetSelectedAssessmentSummaryModelGettersProps) => {
```

Comment:
I think this feels like another place where we could leverage the `AssessmentFunctionalitySwitcher`. This would move these back to deps effectively for the components that had them there prior and might be a bit simpler approach. What do you think?

Reply:
Yes, I think that should work! Let me try that out.